applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -248,16 +372,37 @@ static void stream_configured_cb(struct bt_audio_stream *stream,
 		return;
 	}
 
+	new_pd_us = find_headset_pres_delay(CONFIG_BT_AUDIO_PRES_DLY_SRCH_MODE, channel_index);
+	if (new_pd_us == 0) {
+		LOG_ERR("Cannot get a valid presentation delay");
+	}
+
+	LOG_DBG("New presentation delay: %d us", new_pd_us);
+
+	if (unicast_group) {
 #if CONFIG_STREAM_BIDIRECTIONAL
-	if (ep_state_check(headsets[channel_index].sink_stream.ep,
-			   BT_AUDIO_EP_STATE_CODEC_CONFIGURED) &&
-	    ep_state_check(headsets[channel_index].source_stream.ep,
-			   BT_AUDIO_EP_STATE_CODEC_CONFIGURED))
-#endif /* CONFIG_STREAM_BIDIRECTIONAL */
-	{
-		ret = bt_audio_stream_qos(headsets[channel_index].headset_conn, unicast_group);
-		if (ret) {
-			LOG_ERR("Unable to set up QoS for headset %d: %d", channel_index, ret);
+		if (ep_state_check(headsets[channel_index].sink_stream.ep,
+				   BT_AUDIO_EP_STATE_CODEC_CONFIGURED) &&
+		    ep_state_check(headsets[channel_index].source_stream.ep,
+				   BT_AUDIO_EP_STATE_CODEC_CONFIGURED))
+#endif
+		{
+			if (stream->qos->pd != new_pd_us) {
+				LOG_INF("Changing presentation delay from %dus to %dus",
+					stream->qos->pd, new_pd_us);
+
+				ret = update_sink_stream_qos(new_pd_us);
```

Comment:
Should we update the qos only for the sink stream?

Reply:
Yes

Reply:
OCT-2496 created.